### PR TITLE
KAFKA-17873 Add description to all packages in the public API

### PR DESCRIPTION
**Pull request by m1a2st**
Jira: https://issues.apache.org/jira/browse/KAFKA-17873

In this PR, I add  description about these package
- org.apache.kafka.coordinator.group.api.assignor
- org.apache.kafka.streams.processor.assignment
- org.apache.kafka.streams.processor.assignment.assignors

If there are more kafka api package, we could also add in this PR

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by m1a2st** [comment]
I test in my local, it will show new description
image

File: group-coordinator/group-coordinator-api/src/main/java/org/apache/kafka/coordinator/group/api/assignor/package-info.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Provide the core functionality and metadata management for consumer group partition assignment.
```
Comment (mimaison) [inline group-coordinator/group-coordinator-api/src/main/java/org/apache/kafka/coordinator/group/api/assignor/package-info.java]: All other javadoc start with `Provides`

File: streams/src/main/java/org/apache/kafka/streams/processor/assignment/package-info.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Provides classes and interfaces are used to manage and assign tasks within Kafka Streams applications.
```
Comment (mimaison) [inline streams/src/main/java/org/apache/kafka/streams/processor/assignment/package-info.java]: `are used` -> `used`

**Comment by mimaison** [comment]
@mjsax @ableegoldman Are the proposed descriptions for the Streams packages accurate?

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**